src/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.tsx

```diff
@@ -207,39 +209,48 @@ const LessonDownloadsPage: NextPage<LessonDownloadsPageProps> = ({
   const { onSubmit } = useDownloadForm({ isLegacyDownload: isLegacyDownload });
 
   const onFormSubmit = async (data: DownloadFormProps): Promise<void> => {
-    await debouncedSubmit({
-      data,
-      lessonSlug,
-      setIsAttemptingDownload,
-      setEditDetailsClicked,
-      onSubmit,
-    });
-    const {
-      schoolOption,
-      schoolName,
-      schoolUrn,
-      selectedResourcesForTracking,
-    } = getFormattedDetailsForTracking({
-      school: data.school,
-      selectedResources,
-    });
+    try {
+      await debouncedSubmit({
+        data,
+        lessonSlug,
+        setIsAttemptingDownload,
+        setEditDetailsClicked,
+        setApiError,
```

Comment:
could we not keep `setApiError` contained to this component rather than pass through?

Reply:
ah thanks . Yeah I copied the existing format but it's really nasty when it comes to testing. Have fixed this now.